docs/docs/gatsby-link.md

```diff
@@ -395,7 +395,7 @@ Gatsby's `<Link>` component will only fetch each page's resources once. Updates
 
 In order to prevent this staleness, Gatsby requests an additional resource on each new page load: `app-data.json`. This contains a hash generated when the site is built; if anything in the `src` directory changes, the hash will change. During page loads, if Gatsby sees a different hash in the `app-data.json` than the hash it initially retrieved when the site first loaded, the browser will navigate using `window.location`. The browser fetches the new page and starts over again, so any cached resources are lost.
 
-This only works if the page has not previously been loaded, as the `app-data.json` will not be re-requested for that page. Therefore staleness isn't a solved problem, but it should be a less frequent one.
+However, if the page has previously loaded, it will not re-request `app-data.json`. In that case, the hash comparison will not occur and the previously loaded content will be used. Therefore staleness isn't a solved problem, but it should be a less frequent one.
```

Comment:
```suggestion
However, if the page has previously loaded, it will not re-request `app-data.json`. In that case, the hash comparison will not occur and the previously loaded content will be used. 
```
I believe this is an accurate statement, but it's a bit more editorial than usually included in the docs. It seems as if explaining the potentially challenging case is sufficient?